odoo/addons/base/res/res_config.py

```diff
@@ -492,6 +506,40 @@ def default_get(self, fields):
             if self._fields[name].type == 'selection':
                 res[name] = int(res[name])
 
+        # config: get & convert stored ir.config_parameter (or default)
+        for name, icp in classified['config']:
+            field = self._fields[name]
+            value = SuIrConfigParameter.get_param(icp, field.default(self) if field.default else False)
+            if value is not False:
+                if field.type in ['many2one']:
+                    try:
+                        value = int(value)
+                    except ValueError:
+                        _logger.warning('Unexpected value while converting %s as %s value of field %s.\nValue: %s' % (icp, field.type, field, value))
+                        value = False
+                    # Special case when value is the id of a deleted records, we do not want to
+                    # block the settings screen
+                    if len(self.env[field.comodel_name].sudo().browse(value).exists()) != 1:
```

Comment:
The method `exists()` returns either 0 or 1 record, and does not depend on `sudo()`. This is more intuitive:
```python
    if not self.env[field.comodel_name].browse(value).exists():
```

Reply:
good to know, I fixed it